src/components/[guild]/RolePlatforms/RolePlatforms.tsx

```diff
@@ -33,7 +34,11 @@ type Props = {
 export const openRewardSettingsGuildPlatformIdAtom = atom(0)
 
 const RolePlatforms = ({ roleId }: Props) => {
-  const { onOpen } = useAddRewardContext()
+  const { onOpen, setTargetRoleId } = useAddRewardContext()
+
+  useEffect(() => {
```

Comment:
As far as I can see we don't need a useEffect and a state for this in the provider, we can just pass it as a prop in RolePlatformsWrapper

Reply:
I am not sure about that, the targetRoleId is needed in multiple components, should we prop drill to all of them?

Reply:
No, I mean we can pass it to the provider as a prop and just put in into the context there so it's available in every children, don't need the state and the useEffect for it